mdal/frmts/mdal_esri_tin.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2019 Vincent Cloarec (vcloarec at reos dot site)
+*/
+
+#include "mdal_esri_tin.hpp"
+
+//from https://stackoverflow.com/questions/41012414/convert-double-value-from-little-endian-to-big-endian
+template <typename T>
+static bool readAndChangeEndianness( T &value, std::ifstream &in )
+{
+  char *const p = reinterpret_cast<char *>( &value );
+  in.read( p, sizeof( T ) );
+  if ( in.eof() )
+    return false;
+
+  for ( size_t i = 0; i < sizeof( T ) / 2; ++i )
+    std::swap( p[i], p[sizeof( T ) - i - 1] );
+
+  return true;
+}
+MDAL::DriverEsriTin::DriverEsriTin(): Driver( "ESRI_TIN",
+                                              "Esri TIN",
+                                              "*.adf",
+                                              Capability::ReadMesh
+                                              )
+{}
+
+MDAL::Driver *MDAL::DriverEsriTin::create()
+{
+    return new DriverEsriTin();
+}
+
+std::unique_ptr<MDAL::Mesh> MDAL::DriverEsriTin::load( const std::string &uri, MDAL_Status *status )
+{
+    Vertices vertices;
+    Faces faces;
+
+    populateVertices( uri, vertices );
+    populateFaces( uri, faces );
+
+    std::unique_ptr< MemoryMesh > mesh(
+    new MemoryMesh(
+      name(),
+      vertices.size(),
+      faces.size(),
+      3,
+      computeExtent( vertices ),
+      uri
+    )
+  );
+
+  addTerrainElevationDatasetGroup( mesh.get(), vertices );
+
+  return mesh;
+}
+
+bool MDAL::DriverEsriTin::canRead(const std::string &uri)
+{
+
+    std::string zFileName = zFile( uri );
+    std::string faceFileName = faceFile( uri );
+
+    std::ifstream xyIn( xyFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! xyIn.is_open() )
+        return false;
+
+    std::ifstream zIn( zFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! zIn.is_open() )
+        return false;
+
+    std::ifstream faceIn( faceFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! faceIn.is_open() )
+        return false;
+
+    //check if the file sizes of the nodes are coherent ?
```

Comment:
i think it does not need to be... what you can check is first line of some file if they have some standard header/first line

Reply:
There isn't more information that the position of the points and the indexes of the face. The unused first bytes (64 bytes for tnxy.adf, and 16 bytes for tnz.adf) are used by ArcGIS to store information about 4 virtual points created by ArcGIS for triangulation/generation (https://en.wikipedia.org/wiki/Esri_TIN, see Superpoint). Those points seem to be necessary for ArcGIS but not for MDAL.

So, for me, the only information to know if a file can be read is :
- if it can be opened
- if the size of the tns.adf file (Z value file) is coherent with the size of of tnxy.adf file.
I am not sure of the relevance of checking the size file coherence.

Reply:
lets keep it as it is, just remove commented out code please